modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java

```diff
@@ -566,11 +602,19 @@ public void processOpts() {
         } else if (SERIALIZATION_LIBRARY_GSON.equals(getSerializationLibrary())) {
             additionalProperties.put(SERIALIZATION_LIBRARY_GSON, "true");
             additionalProperties.remove(SERIALIZATION_LIBRARY_JACKSON);
+            additionalProperties.remove(SERIALIZATION_LIBRARY_JSONB);
```

Comment:
this is the place where you should use switch/case for these instead adding another `else if`